src/services/middlewareServices/AuthenticationMiddlewareService.ts

```diff
@@ -80,7 +153,12 @@ export default class AuthenticationMiddlewareService {
       const accessToken = retrievedToken
         ? jwtUtils.decryptToken(retrievedToken)
         : ""
-      return { accessToken, githubId, isomerUserId, email }
+      return {
+        accessToken,
+        githubId: githubId ?? "",
```

Comment:
Hmm why are we returning an empty string here ya?

Reply:
the type definition is `SessionDataProps = IsomerUserProps | GithubUserProps`, which expands into 

```ts
type IsomerUserProps = {
  isomerUserId: string
  email: string
}

type GithubUserProps = IsomerUserProps & {
  githubId: string
  accessToken: string
}

type ConcreteProps = {
  githubId?: string
  accessToken?: string
  isomerUserId: string
  email: string
}
```

Reply:
i can remove the `""` on `githubId` but we need it for `email`

Reply:
hmm if `githubId` not required, what do we feel about making them optional in the interface?

Reply:
more inclined to raise a separate PR for this if required tbh

Reply:
nah is fine, may I get your help to add a small comment summarising this limitation?

Reply:
done